fidesctl/src/fideslang/models.py

```diff
@@ -92,6 +94,7 @@ class DatasetField(BaseModel):
     data_qualifier: FidesKey = Field(
         default="aggregated.anonymized.unlinked_pseudonymized.pseudonymized.identified",
     )
+    fields: Optional[List[DatasetField]]
```

Comment:
i think this might not be the right place for this, but I could totally be wrong. my instinct would be to update `collections` so that it can take either another collection or fields, instead of assuming fields could have more fields. This way, fields stay as the lowest possible level `always`, and we allow collections themselves to be nested

Reply:
so for `collection` you would change is to `fields: List[DatasetCollection|DatasetField]

does that make sense? I'm not actually sure if it would work (referencing a model from within itself) but it might be a better paradigm. That being said, if the current model works just fine then don't rework everything just for this!

Reply:
@ThomasLaPiana when starting off I was thinking it could go either way, but I can see that being a valuable change for sure 🤔 

The reason I stuck with `fields` being nested over `collections` was partly due to chatting with the fidesops team and [their research in here to use nested fields](https://ethyca.atlassian.net/wiki/spaces/OSS/pages/2111897639/NoSql+address+format). Additionally it appears that with mongodb at least collections are not nested so wanted to avoid potential confusion (looking at [our docs](https://ethyca.github.io/fides/tutorial/dataset/#understanding-the-dataset-resource) as well as [mongodb docs](https://docs.mongodb.com/compass/current/schema/export/#:~:text=The%20example%20above%20outputs%20the%20following%20schema%3A))

I still haven't gone through the `annotate-dataset` command yet outside of the review in #310 so am going to look there and see if that helps make a decision easier, let me know if any of the above context has you lean one way or another more strongly 👍🏽

Reply:
I didn't see your second comment somehow before adding my response 🤦🏽  but I _think_ what you suggested should also work🤞🏽 as far as self-referencing - will validate the current state through the dataset annotation command as well so we can make a good decision either way :)

Reply:
> > @SteveDMurphy its getting fixed in another PR, but have you considered how this might break `annotate-dataset`? it has a hard-coded loop when going through datasets 👀
> 
> Thanks @ThomasLaPiana ! Will take a look this morning. I had not gotten there yet after our chicken/egg conversation around the `generate-dataset` command last week and how that would look, deferring until later and focusing this one on the `evaluate` functionality. I can (now) see how those two don't necessarily need to be treated in the same way 👍🏽

@ThomasLaPiana - I reorganized the `get_all_level_fields` generator into `core/utils.py` and confirmed that it will also allow us to annotate nested fields once the fixes in your PR ship

Reply:
ok, if you came to this after collaborating with fidesops team then lets definitely keep it the same. thanks!